code/parachain/frame/staking-rewards/src/test/mod.rs

```diff
@@ -124,12 +128,55 @@ fn stake_in_case_of_low_balance_should_not_work() {
 	});
 }
 
+#[test]
+fn split_doesnt_cause_loss_in_assets() {
+	new_test_ext().execute_with(|| {
+		process_and_progress_blocks::<StakingRewards, Test>(1);
+
+		mint_assets([ALICE], [PICA::ID], PICA::units(10_000));
+
+		create_rewards_pool_and_assert(RewardRateBasedIncentive {
+			owner: ALICE,
+			asset_id: PICA::ID,
+			start_block: 2,
+			end_block: 5,
+			reward_configs: default_reward_config(),
+			lock: default_lock_config(),
+			share_asset_id: XPICA::ID,
+			financial_nft_asset_id: STAKING_FNFT_COLLECTION_ID,
+		});
+
+		process_and_progress_blocks::<StakingRewards, Test>(1);
+
+		let amount = 100_000;
+		let ratio = Permill::from_parts(555_555);
+
+		// 0.555_555 * 100_000 is not an integer, there will be rounding when calculating the
```

Comment:
I wonder if it makes sense to build some utility function around this pattern to avoid mistakes.

Reply:
Resolving as per conversation in slack; came to the conclusion that while it would be nice, it might be more work than it's worth to refactor and use a helper like this